torch/testing/asserts.py

```diff
@@ -0,0 +1,389 @@
+        check_device (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` live in the same
+            :attr:`~torch.Tensor.device` memory. If this check is disabled **and** :attr:`a` and :attr:`b` do not live
+            in the same memory :attr:`~torch.Tensor.device`, they are moved to the CPU memory.
+        check_dtype (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same
+            :attr:`~torch.Tensor.dtype`. If this check is disabled **and** :attr:`a` and :attr:`b` do not have the same
+            :attr:`~torch.Tensor.dtype`, the comparison :attr:`~torch.Tensor.dtype` is determined by
+            :func:`torch.promote_types`.
+        check_stride (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same
+            :meth:`~torch.Tensor.stride`.
+
+    Raises:
+        AssertionError: If :attr:`a` and :attr:`b` do not have the same :attr:`~torch.Tensor.shape`.
+        AssertionError: If :attr:`check_device`, but :attr:`a` and :attr:`b` do not live in the same
+            :attr:`~torch.Tensor.device` memory.
+        AssertionError: If :attr:`check_dtype`, but :attr:`a` and :attr:`b` do not have the same
+            :attr:`~torch.Tensor.dtype`.
+        AssertionError: If :attr:`check_stride`, but :attr:`a` and :attr:`b` do not have the same
+            :meth:`~torch.Tensor.stride`.
+        AssertionError: If the values of :attr:`a` and :attr:`b` are equal up to the desired precision.
+        UsageError: If :attr:`a` or :attr:`b` is complex, quantized, or sparse. This is a temporary restriction and
+            will be relaxed in the future.
+
+
+    The following table displays the default ``rtol`` and ``atol`` for floating point :attr:`~torch.Tensor.dtype`'s.
+    For integer :attr:`~torch.Tensor.dtype`'s, ``rtol = atol = 0.0`` is used.
+
+    +===========================+============+==========+
+    | :class:`~torch.dtype`     | ``rtol``   | ``atol`` |
+    +===========================+============+==========+
+    | :attr:`~torch.float16`    | ``1e-3``   | ``1e-5`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.bfloat16`   | ``1.6e-2`` | ``1e-5`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.float32`    | ``1.3e-6`` | ``1e-5`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.float64`    | ``1e-7``   | ``1e-7`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.complex32`  | ``1e-3``   | ``1e-5`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.complex64`  | ``1.3e-6`` | ``1e-5`` |
+    +---------------------------+------------+----------+
+    | :attr:`~torch.complex128` | ``1e-7``   | ``1e-7`` |
+    +---------------------------+------------+----------+
+
+    .. seealso::
+
+        To compare tensors for bitwise equality, :func:`assert_tensors_equal` can be used.
```

Comment:
"To assert that the values in two tensors are bitwise equivalent use :func:\`assert_tensors_equal\`, instead."